molecule/src/structure_checker2.cpp

```diff
@@ -0,0 +1,516 @@
+/****************************************************************************
+ * Copyright (C) from 2009 to Present EPAM Systems.
+ *
+ * This file is part of Indigo toolkit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+#include "molecule/structure_checker2.h"
+#include "molecule/molecule.h"
+#include "molecule/molecule_automorphism_search.h"
+#include "molecule/molecule_exact_matcher.h"
+#include "reaction/base_reaction.h"
+#include <algorithm>
+#include <functional>
+#include <iterator>
+#include <memory>
+#include <numeric>
+#include <string>
+
+using namespace indigo;
+
+StructureChecker2::StructureChecker2()
+{
+}
+
+bool StructureChecker2::CheckResult::isEmpty() const
+{
+    return messages.size() == 0;
+}
+
+static void message(StructureChecker2::CheckResult& result, StructureChecker2::CheckMessageCode code, int index, std::vector<int> ids,
+                    const StructureChecker2::CheckResult& subresult)
+{
+    StructureChecker2::CheckMessage m = {code, index, ids, subresult};
```

Comment:
Unfortunately, 2 of 3 Indigo CI checks fail at this line.

### build_mac/Build native libs:
```
/Users/runner/work/Indigo/Indigo/molecule/src/structure_checker2.cpp:45:37: error: no matching constructor for initialization of 'StructureChecker2::CheckMessage'
     StructureChecker2::CheckMessage m = {code, index, ids, subresult};
                                     ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/runner/work/Indigo/Indigo/molecule/../molecule/structure_checker2.h:109:26: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 4 were provided
```
### build_linux/Build native libs:
```
/__w/Indigo/Indigo/molecule/src/structure_checker2.cpp:45:69: error: could not convert '{code, index, ids, subresult}' from '<brace-enclosed initializer list>' to 'indigo::StructureChecker2::CheckMessage'
   45 |     StructureChecker2::CheckMessage m = {code, index, ids, subresult};
      |                                                                     ^
      |                                                                     |
      |                                                                     <brace-enclosed initializer list>
```
Possible reason can be due to inline initialization in the class: 
```
        struct DLLEXPORT CheckMessage
        {
            CheckMessageCode code = StructureChecker2::CheckMessageCode::CHECK_MSG_NONE;
...
            int index = -1;
```

See discussion: https://stackoverflow.com/questions/37776823/could-not-convert-from-brace-enclosed-initializer-list-to-struct

Reply:
Fixed